subworkflows/local/align_star/main.nf

```diff
@@ -62,6 +62,16 @@ workflow ALIGN_STAR {
     BAM_SORT_STATS_SAMTOOLS ( ch_orig_bam, fasta )
     ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)
 
+    publish:
+    ch_orig_bam         >> (params.save_align_intermeds || params.save_umi_intermeds ? 'star_salmon/' : null)
```

Comment:
This is fine for specific pipelines, but makes module reusability across pipelines more cumbersome. If we need to have ternary operators in every nf-core module to control whether an output is published, this would make the pipeline schema, potentially huge.

Reply:
I agree, I cut some corners here by not passing the params as workflow inputs. In the final implementation I would do that so that you can choose whether to expose it as a param in your own pipeline